lib/krb5/principal.c

```diff
@@ -348,6 +352,8 @@ krb5_parse_name_flags(krb5_context context,
     }
     if (enterprise)
 	(*principal)->name.name_type = KRB5_NT_ENTERPRISE_PRINCIPAL;
+    else if (srv_and_inst)
```

Comment:
We can avoid the srv_and_inst variable entirely and use a comparison of 

   if (ncomp > 1 && strcmp(comp[0], KRB5_TGS_NAME) == 0)

I would feel safer with the comparison being case sensitive.
What do you think?